packages/pyright-internal/src/analyzer/checker.ts

```diff
@@ -2484,7 +2484,15 @@ export class Checker extends ParseTreeWalker {
 
         localTypeVarUsage.forEach((usage) => {
             // Report error for local type variable that appears only once.
+            // Exempt the case where the single occurrence is solely in the return type
+            // annotation (e.g., def f[T]() -> T), which is considered valid.
             if (usage.nodes.length === 1 && !usage.isExempt) {
+                const onlyInReturn = usage.returnTypeUsageCount === 1 && usage.paramTypeUsageCount === 0;
+                // TODO: this check doesn't work with function type comments, this is a bug with the collection
+                //  def f(x):  # type: (T) -> None
```

Comment:
want me to investigate this?

Reply:
seems like a waste of time, i definitely wouldn't

Reply:
i figured out how to partially fix it. it fixes the false negative in that example and in the broken test (`# type: (T) -> None`) but not the false positive if you do `# type: () -> T`

i agree spending any more time on this is a waste of time, so i think this is good enough